### ISSUE TITLE
Jakarta Data: prefix WLP is prepended to table names by default

### ISSUE THREAD

**Issue by tjwatson** [OpenLiberty/open-liberty#28995]
**Describe the bug**  
`WLP` default prefixed is used when using the default `dataStore` for the `@Repository`.  

I have an InstantOn test that shows this in my PR https://github.com/OpenLiberty/open-liberty/pull/28979. There I had to use wlpusers as the table name when initializing the database with [test data](https://github.com/OpenLiberty/open-liberty/pull/28979/files#diff-195a464afe35f4a658260133f8a0a73131b95f684f09bab31063b671d1a658efR65-R80)

This was unexpected to me when porting over a Spring Data test [UserRepository](https://github.com/OpenLiberty/open-liberty/blob/78150fb8f77af9b53e708412f21c5564257a7f1c/dev/io.openliberty.checkpoint.springboot.fat30.data.app/src/main/java/io/openliberty/checkpoint/springboot/fat30/data/repositories/UserRepository.java#L20) where there was no prefix added to the table name in the [test data](https://github.com/OpenLiberty/open-liberty/blob/78150fb8f77af9b53e708412f21c5564257a7f1c/dev/io.openliberty.checkpoint_fat_springboot_data/fat/src/io/openliberty/checkpoint/fat/springboot/data/FATSuite.java#L77-L87)

**Comment by njr-11** [comment]
I removed the `bug` label and added this under the configuration user story for Jakarta Data.  It isn't broken.  A design decision needs to be made about which defaults to use. Currently, if you don't specify the dataStore for `@Respository` you end up with the defaultDatabaseStore (which has the WLP table prefix) and the DefaultDataSource.  The Jakarta Data spec requires the DefaultDataSource part, but we could choose differently regarding the table prefix if we think it is more desirable to not have one.

It should be noted we cannot remove the WLP prefix from `defaultDatabaseStore` because that is existing Liberty default configuration (currently used as a default for Persistent EJB Timers), but we could create a different databaseStore with the table prefix configuration absent.

**Comment by tjwatson** [comment]
Can you configure the `defaultDatabaseStore` in the server.xml to specify an empty prefix?

**Comment by njr-11** [comment]
> Can you configure the `defaultDatabaseStore` in the server.xml to specify an empty prefix?

Yes, `<databaseStore id="defaultDatabaseStore" tablePrefix=""/>`

**Comment by tjwatson** [comment]
> Yes, `<databaseStore id="defaultDatabaseStore" tablePrefix=""/>`

I confirmed that works in my test.  That gives us better parity for migrating from Spring Data.  It is not too arduous, but we may still want to have defaults that allow a more simple migration from Spring Data.

**Comment by njr-11** [comment]
Jakarta Data should no longer be adding table prefixes unless the user configures a databaseStore that indicates to use one.
